### PR TITLE
fix build error on QueryProfiler

### PR DESCRIPTION
**Pull request by MaxWk**
Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
n/a

Detailed description / Documentation draft:
fix build error: https://github.com/ClickHouse/ClickHouse/issues/32210

> By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.

> If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide first.

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Common/QueryProfiler.cpp [hunk 1]
```diff
@@ -31,6 +31,8 @@ namespace
     /// And so to overcome this flag is exists,
     /// to ignore delivered signals after timer_delete().
     thread_local bool signal_handler_disarmed = true;
+#else
+    thread_local bool signal_handler_disarmed = false;
```
Comment (azat) [inline src/Common/QueryProfiler.cpp:35]: Let's not create unused variable for osx, but instead, fix it in the `QueryProfilerBase<ProfilerImpl>::tryCleanup()`.

And speaking about `QueryProfiler`, AFAICS `timer_t`/`timer_create`/`timer_settime`/`timer_delete` is not exists in osx, and it should complain on it too, no?

Comment (azat) [reply src/Common/QueryProfiler.cpp:35]: Ok, let's simply move `signal_handler_disarmed` out of the `#if defined(OS_LINUX)`

Comment (azat) [reply src/Common/QueryProfiler.cpp:35]: Here it is - https://github.com/ClickHouse/ClickHouse/pull/32219/commits/37ae9f943013448b42b57d0527adcbc40bcb52ac

P.S. sorry I cannot push into your topic branch, hence in a separate PR.

Comment (MaxWk) [reply src/Common/QueryProfiler.cpp:35]: that's ok ,just slove the problem

**Pull request closed without merging**